### PR TITLE
[14.0] product_assortment: misc fixes

### PR DESCRIPTION
**Pull request by Kev-Roche**

### CODE REVIEW

**Comment by Kev-Roche** [comment]
This PR highlights and fixes 2 different issues : 
1) error when creating and saving a new assortment.
2) error when adding new product in black or whitelist.

Linked to issue #870

**Review by rousseldenis — APPROVED** [review]
LGTM

**Comment by Kev-Roche** [comment]
cc @florian-dacosta, @hparfr

**Review by florian-dacosta — APPROVED** [review]

**Review by hparfr — APPROVED** [review]

File: product_assortment/models/ir_filters.py [hunk 1]
```diff
@@ -64,7 +64,10 @@ def _get_eval_domain(self):
 
         return res
 
-    @api.depends("domain", "blacklist_product_ids", "whitelist_product_ids")
+    # Original : @api.depends("domain", "blacklist_product_ids", "whitelist_product_ids")
+    # TODO : Resolve bug if depending of blacklist_product_ids and
```
Comment (simahawk) [inline product_assortment/models/ir_filters.py]: is this TODO to be solved now?

Comment (Kev-Roche) [reply product_assortment/models/ir_filters.py]: No, it s working without. And sorry about my PR name, it was not very clear what it is about.

Comment (simahawk) [reply product_assortment/models/ir_filters.py]: but these fields are mandatory for computing the domain. Could you be more explicit on the error? The issue you linked does not explain the reason for this.

Comment (rousseldenis) [reply product_assortment/models/ir_filters.py]: @Kev-Roche

Comment (Kev-Roche) [reply product_assortment/models/ir_filters.py]: It s an Access Error happending when saving a product in black/whitelist:
"Database Fetch misses ids ((< NewID origin=12 >,)) and has extra ids ((12,)), may be caused by a type incoherence in a previous request".

**Comment by Bart-dh** [comment]
Looks good! Much needed fix

**Review by Bart-dh — APPROVED** [review]

File: product_assortment/models/ir_filters.py (hunk 1 above)
Comment (xavier-bouquiaux) [reply product_assortment/models/ir_filters.py]: Hi @Kev-Roche 
I've included your PR in my project and i still have this kind of issue as soon as i use whitelist/blacklist 
Database fetch misses ids ((<NewId origin=49486>,)) and has extra ids ((49486,)), may be caused by a type incoherence in a previous request

Comment (Cedric-Pigeon) [reply product_assortment/models/ir_filters.py]: @Kev-Roche I just pushed a PR on your repo fixing the problem: https://github.com/akretion/product-attribute/pull/13
The depends is not needed as the field is not stored. If you depends on the domain, the widget "domain" disturb the cache.

Comment (Kev-Roche) [reply product_assortment/models/ir_filters.py]: @Cedric-Pigeon Thank you for your PR and finding the real problem.

Comment (Kev-Roche) [reply product_assortment/models/ir_filters.py]: @Cedric-Pigeon , it's weird, a test failed with your PR

Comment (Cedric-Pigeon) [reply product_assortment/models/ir_filters.py]: @Kev-Roche yes it is only a cache issue. Adding self.assortment.invalidate_cache() before line 106 should fix the test

**Comment by dreispt** [comment]
Anyone against a merge here?

**Comment by hparfr** [comment]
It looks like it's ready to be merged.

**Pull request merged**